src/payment/bolt11.rs

```diff
@@ -254,62 +261,183 @@ impl Bolt11Payment {
 		}
 	}
 
+	/// Allows to attempt manually claiming payments with the given preimage that have previously
+	/// been registered via [`receive_for_hash`] or [`receive_variable_amount_for_hash`].
+	///
+	/// This should be called in reponse to an [`PaymentClaimable`] event.
+	///
+	/// Will check that the payment is know, and that the given preimage and claimable amount match
+	/// our expectations before attempting to claim the payment.
+	///
+	/// Upon success, a [`PaymentReceived`] event will be emitted.
+	///
+	/// [`receive_for_hash`]: Self::receive_for_hash
+	/// [`receive_variable_amount_for_hash`]: Self::receive_variable_amount_for_hash
+	/// [`PaymentClaimable`]: crate::Event::PaymentClaimable
+	/// [`PaymentReceived`]: crate::Event::PaymentReceived
+	pub fn claim_manually(
```

Comment:
We should add a way to quickly fail the payment if the preimage cannot be obtained, rather than having to wait for invoice expiration.